### PR TITLE
server: fix span "server.dispatch" not passed to the context

### LINKED ISSUE #23579: most of the opentracing span not created due to the root span not passed to the context.
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

I tested tidb with opentracing enabled, using opentelemetry-collector to collect traces. everything looks good but only 'server.dispatch' span was captured. I have already read some of the code, it looks like other types of span never created when tidb is running. I found some code in **server/conn.go** , span 'server.dispatch' created, but not pass to the context. I changed some code and the bug fixed.

### 2. What did you expect to see? (Required)

tracing details when sql executes, in other APM system like skywalking, jager etc.

### 3. What did you see instead (Required)

only "server.dispatch" span can be captured

### 4. What is your TiDB version? (Required)

release-4.0
master

### PR DESCRIPTION
**Pull request by RickyWuCN**
### What problem does this PR solve?

Issue Number: close #23579

Problem Summary:

### What is changed and how it works?

What's Changed: pass the span to the context, so other opentracing related codes could work.

How it Works: 

### Related changes

- None

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)
I compiled and tested with open-telemetry collector, and I store all spans in aliyun for testing.

Side effects

- None

### Release note 

- fixed some opentracing problem so more details can be collected

### CODE REVIEW

File: server/conn.go [hunk 1]
```diff
@@ -942,6 +942,7 @@ func (cc *clientConn) dispatch(ctx context.Context, data []byte) error {
 	}
 
 	span := opentracing.StartSpan("server.dispatch")
+	ctx = opentracing.ContextWithSpan(ctx, span)
```
Comment (zz-jason) [inline server/conn.go:945]: could you add some tests to ensure the fix takes effect?

**Comment by morgo** [comment]
I believe this was changed intentionally due to a performance regression. See: https://github.com/pingcap/tidb/pull/23326

/cc @tiancaiamao

**Comment by RickyWuCN** [comment]
yes, that looks better, and this can be closed.

**Pull request closed without merging**